runtime/lua/vim/lsp/util.lua

```diff
@@ -1777,25 +1777,33 @@ end
 
 local str_utfindex = vim.str_utfindex
 ---@private
-local function make_position_param()
+local function make_position_param(offset_encoding)
```

Comment:
should this be required?

    assert(offset_encoding ~= nil)

Reply:
I think `client` is better

Reply:
Can clients attach to the same buffer have different `offset_encoding`?

Reply:
> Can clients attach to the same buffer have different `offset_encoding`?

probably, but as a first pass we can just document that this isn't supported. The general idea you have here should solve the common case.

Reply:
> > Can clients attach to the same buffer have different `offset_encoding`?
> 
> probably, but as a first pass we can just document that this isn't supported. The general idea you have here should solve the common case.

Also as part of this PR, it should be not much work and will avoid confusion to: 

1. add a check in the client-attach logic, that compares the offset_encoding of all other clients attached to the same buffer. Show a warning (or even an error) if the offset_encoding differs amongst clients attached to the same buffer.